docs/root/configuration/http/cluster_specifier/lua.rst

```diff
@@ -0,0 +1,45 @@
+.. _config_http_cluster_specifier_lua:
+
+Lua cluster specifier
+========================
+
+Overview
+--------
+
+The HTTP Lua cluster specifier allows `Lua <https://www.lua.org/>`_ scripts to select router cluster
+during the request flows. `LuaJIT <https://luajit.org/>`_ is used as the runtime. Because of this, the
+supported Lua version is mostly 5.1 with some 5.2 features. See the `LuaJIT documentation
+<https://luajit.org/extensions.html>`_ for more details.
+
+Configuration
+-------------
+
+* This filter should be configured with the type URL ``type.googleapis.com/envoy.extensions.router.cluster_specifiers.lua.v3.LuaConfig``.
+* :ref:`v3 API reference <envoy_v3_api_msg_extensions.router.cluster_specifiers.lua.v3.LuaConfig>`
+
+A simple example of configuring Lua cluster specifier is as follow:
+
+.. code-block:: yaml
+
+  name: envoy.router.cluster_specifier_plugin.lua
+  typed_config:
+    "@type": type.googleapis.com/envoy.extensions.router.cluster_specifiers.lua.v3.LuaConfig
+    source_code:
+      inline_string: |
+        function envoy_on_cluster(header_handle)
```

Comment:
Rather than `envoy_on_cluster`, it would be better to call it `envoy_on_route` because it works when doing route selecting.

Reply:
And it would be better to provide a handle like http lua filter. The handle will expose a `headers` method. The we can provide more context by the handle in the future.

And please provide the API doc to the handle and http header map. See https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/lua_filter

Reply:
done, thanks.